sklearn/calibration.py

```diff
@@ -25,6 +26,197 @@
 from .svm import LinearSVC
 from .model_selection import check_cv
 from .metrics.classification import _check_binary_probabilistic_predictions
+from .metrics.pairwise import euclidean_distances
+from .metrics.ranking import roc_curve
+
+
+class OptimalCutoffClassifier(BaseEstimator, ClassifierMixin):
+    """Optimal cutoff point selection.
+
+    If cv="prefit" the base estimator is assumed to be fitted and all data will
+    be used for the selection of the cutoff point that determines the
+    output of predict. Otherwise predict will use the average of the thresholds
+    resulting from the cross-validation loop.
+
+    Parameters
+    ----------
+    base_estimator : instance BaseEstimator
+        The classifier whose decision threshold will be adapted according to the
+        acquired optimal cutoff point
+
+    method : str
+        The method to use for choosing the optimal cutoff point.
+
+        - 'roc', selects the point on the roc_curve that is closer to the ideal
+        corner (0, 1)
+
+        - 'max_se', selects the point that yields the highest sensitivity with
```

Comment:
I prefer the terms "true positive rate" and "true negative rate" because they are pretty clear in their meaning.